lib/src/accordion/accordion.component.spec.ts

```diff
@@ -0,0 +1,109 @@
+import { async, ComponentFixture, TestBed } from "@angular/core/testing";
+import { By } from "@angular/platform-browser";
+import { AccordionComponent, AccordionIconRotation, AccrodionListItem } from "./accordion.component";
+import { SafeHtmlPipe } from "./accordion.pipe";
+import { DynamicStylePipe } from "./accordion-style.pipe";
+import { CommonModule } from "@angular/common";
+
+@Component({
+    selector: "test-sebng-textboxgroup",
+    template: `
+        <sebng-accordion
+            [className]="className"
+            [customIcon]="customIcon"
+            [customIconExpanded]="customIconExpanded"
+            [iconPosition]="iconPosition"
+            [iconRotation]="iconRotation"
+            [id]="id"
+            [list]="list"
+            [alternative]="alternative"
+            [activeIndex]="activeIndex"
+        ></sebng-accordion>
+    `,
+})
+class AccordionTestComponent {
+    className?: string;
+    customIcon?: string;
+    customIconExpanded?: string;
+    iconPosition?: "left" | "right";
+    iconRotation?: AccordionIconRotation;
+    id?: string;
+    list: Array<AccrodionListItem>;
+    alternative?: boolean;
+    activeIndex?: number;
+}
+
+describe("AccordionComponent", () => {
+    let component: AccordionTestComponent;
+    let fixture: ComponentFixture<AccordionTestComponent>;
+
+    const accordionList: Array<AccrodionListItem> = [
+        { header: "Item 1", content: { title: "title", desc: "desc" } },
+        {
+            header: "Item 2",
+            content: [
+                { title: "title", desc: "desc" },
+                { title: "title", desc: "desc" },
+            ],
+        },
+        { header: "Item 3", content: [{ desc: "desc" }, { desc: "desc" }] },
+    ];
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [CommonModule],
+            declarations: [AccordionComponent, AccordionTestComponent, SafeHtmlPipe, DynamicStylePipe],
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(AccordionTestComponent);
+        component = fixture.componentInstance;
+        component.list = accordionList;
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+        expect(fixture.debugElement.query(By.css(".text-wrapper"))).toBeTruthy();
+    });
+
+    it("Should render custom className and id", () => {
+        const className: string = "myAccordionClass";
+        const id: string = "myAccordionId";
+
+        component.id = id;
+        component.className = className;
+
+        fixture.detectChanges();
+        expect(fixture.debugElement.query(By.css(`.${className}`))).toBeTruthy();
+        expect(fixture.debugElement.query(By.css(`#${id}`))).toBeTruthy();
+    });
+
+    it("Should render subheader is included in props", () => {
+        const newAccordionList = JSON.parse(JSON.stringify(accordionList));
```

Comment:
```suggestion
        const newAccordionList: Array<AccrodionListItem> = JSON.parse(JSON.stringify(accordionList));
```

Reply:
Also use [deepCopy](https://github.com/sebgroup/frontend-tools/wiki/deepCopy) from our frontend-tools instead of JSON manipulation

Reply:
Is that really necessary ? It's angular and unit test at that .

Reply:
yep, it's necessary to catch any changes we make on the APIs